mama/c_cpp/src/c/mama/publisher.h

```diff
@@ -196,8 +283,61 @@ extern mama_status
 mamaPublisher_getTransport (mamaPublisher   publisher,
                             mamaTransport*  transport);
 
+/**
+ * Allocate a publisher events callback structure.
+ * This is done to allow the structure to change more easily.
+ * This structure is disposed by calling mamaPublisherCallbacks_deallocate.
+ *
+ * @param cb Pointer to pointer to a structure.
+ */
+MAMAExpDLL
+extern mama_status
+mamaPublisherCallbacks_allocate (mamaPublisherCallbacks** cb);
+
+/**
+ * Deallocate a publisher events callback structure.
+ * This structure was obtained by calling mamaPublisherCallbacks_allocate.
+ *
+ * @param cb Pointer to a structure.
+ */
+MAMAExpDLL
+extern mama_status
+mamaPublisherCallbacks_deallocate (mamaPublisherCallbacks* cb);
+
+/**
+ * Retrieve the publisher callbacks.
+ *
+ * @param publisher The mamaPublisher from which to send the response.
+ * @param cb Pointer to a cb structure to be populated with the current callbacks.
+ */
+MAMAExpDLL
+extern mama_status
+mamaPublisher_getUserCallbacks (mamaPublisher publisher,
```

Comment:
Same as above this is for parameter alignment. I know if can be painful because a lot of these files contain mixed-standards but the double-column approach is the one we try and use for any new code where all arguments won't fit on a single line (or optionally even if they would).